profiles/views_test.py

```diff
@@ -0,0 +1,136 @@
+"""
+Tests for profile view
+"""
+import json
+
+from dateutil.parser import parse
+from django.contrib.auth.models import User
+from django.core.urlresolvers import reverse
+from django.test import TestCase
+from rest_framework.status import (
+    HTTP_403_FORBIDDEN,
+    HTTP_405_METHOD_NOT_ALLOWED,
+)
+
+from profiles.factories import ProfileFactory
+from profiles.models import Profile
+from profiles.serializers import (
+    ProfileLimitedSerializer,
+    ProfilePrivateSerializer,
+    ProfileSerializer,
+)
+
+
+class ProfileTests(TestCase):
+    """
+    Tests for GET on profile view
+    """
+
+    def setUp(self):
+        """
+        Create a user and profile
+        """
+        super(ProfileTests, self).setUp()
+
+        self.user1_credentials = {
+            "username": "user1",
+            "password": "pass"
+        }
+        self.user1 = User.objects.create_user(**self.user1_credentials)
+        self.url1 = reverse('profile-detail', kwargs={'user': self.user1.username})
+
+        self.user2_credentials = {
+            "username": "user2",
+            "password": "pass"
+        }
+        self.user2 = User.objects.create_user(**self.user2_credentials)
+        self.url2 = reverse('profile-detail', kwargs={'user': self.user2.username})
+
+    def test_no_anonymous_access(self):
```

Comment:
Ideally, this test would be a simple assertion that the permission classes are set correctly (is authenticated and the edit one you made above). If that's true and they're both tested to work.. then DRF will handle this for us and we don't have to make (slow) integration tests for it. DRF already tests that functionality for us.